File: Java/Spring Boot/ecza-crm/src/main/java/com/farukecza/eczacrm/entity/Product.java
package com.farukecza.eczacrm.entity;

import lombok.Data;
import lombok.Getter;

import javax.persistence.*;

@Entity
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false)
    private String name;
    private int producerId;
    @ManyToOne(optional = false)
    @JoinColumn(name = "category_id",foreignKey = @ForeignKey(name="FK_CATEGORY_PRODUCT_CATEGORY"))
    private Category category;
    private double price;
}

File: Scala/MyFirtScalaApp/src/product/Product.scala
package product

class Product(private var _name: String, val price: Float) {

  def name = _name

  def name_=(newValue: String) = {
    _name = newValue
  }
  override def toString: String = this.name
}

object Product{
  def apply (name:String,price:Float):Product = new Product(name,price)
}